repos/fdbt-site/src/pages/api/deleteProduct.ts

```diff
@@ -0,0 +1,22 @@
+import { NextApiResponse } from 'next';
+import { deleteProductByNocCodeAndId } from '../../data/auroradb';
+import { redirectToError, redirectTo, getAndValidateNoc } from '../../utils/apiUtils';
+import { NextApiRequestWithSession } from '../../interfaces';
+
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
+    try {
+        const id = Number(req.query?.id);
+
+        if (!Number.isInteger(id)) {
+            throw new Error('insufficient data provided for delete query');
+        }
+
+        const nationalOperatorCode = getAndValidateNoc(req, res);
+
+        await deleteProductByNocCodeAndId(id, nationalOperatorCode);
+
+        redirectTo(res, req.headers.referer ?? '/products/services');
```

Comment:
I think we usually have a return after a redirect in an API - unsure if its specifically needed here as its the last line?

Reply:
```suggestion
        redirectTo(res, req.headers.referer ?? '/products/services');
        return;
```

Reply:
As discussed